drivers/vhm/vhm_hugetlb.c

```diff
@@ -225,6 +225,47 @@ void hugepage_free_guest(struct vhm_vm *vm)
 	mutex_unlock(&vm->hugepage_lock);
 }
 
+u64 hugepage_gpa_to_hpa(struct vhm_vm *vm, u64 guest_phys)
+{
+	struct hugepage_map *map;
```

Comment:
Would below code seem more readable ?
u64 hpa = 0ULL;

if (guest_phys > map->guest_gpa + map->size)
	break;

	hpa = map->vm0_gpa + guest_phys - map->guest_gpa;
	goto done;
}

done:
	mutex_unlock(&vm->hugepage_lock);
        if (!hpa)
	        printk(KERN_WARNING "cannot find correct mem map, please check the "
		         "input's range or alignment");
        return hpa;